perfkitbenchmarker/providers/ibmcloud/util.py

```diff
@@ -0,0 +1,212 @@
+}
+
+
+def ReadConfig(config):
+  """Reads in config yml
+
+  Args:
+    config: config file name.
+
+  Returns:
+    The json representation of the config file.
+  """
+  try:
+    with open(config, 'r') as stream:
+      data = json.dumps(yaml.load(stream), sort_keys=True)
+      return json.loads(data)
+  except Exception as ex:
+    raise errors.Error('Failed to load configuration file %s, %s', config, ex)
+
+
+def GetSubnetIndex(ipv4_cidr_block):
+  """Finds the index for the given cidr
+
+  Args:
+    ipv4_cidr_block: cidr to find.
+
+  Returns:
+    The index number of the found cidr as in the predefined list
+  """
+  for name in SUBNETXS:
+    ip_list = SUBNETS_EXTRA[name]
+    for i in range(len(ip_list)):
+      if ip_list[i] == ipv4_cidr_block:
+        return i
+  return -1
+
+
+def GetRouteCommands(data, index, target_index):
+  """Creates a list of ip route commands in text format to run on vm,
+    not used on normal perfkit runs.
+
+  Args:
+    data: output from route command on vm.
+    index: subnet index.
+    target_index: target subnet index.
+
+  Returns:
+    The index number of the found cidr as in the predefined list
+  """
+  route_cmds = []
+  if data:
+    for subnet_name in SUBNETXS:
+      subnet_cidr_block = SUBNETS_EXTRA[subnet_name][index]
+      target_cidr_block = SUBNETS_EXTRA[subnet_name][target_index]
+      subnet_gateway = SUBNETS_EXTRA_GATEWAY[subnet_name][index]
+      route_entry = subnet_cidr_block.split('/24')
+      interface = None
+      for line in data.splitlines():
+        items = line.split()
+        if len(items) > 6 and items[0] == route_entry[0]:
+          interface = items[7]
+          route_cmds.append('ip route add ' + target_cidr_block + ' via ' + \
+                            subnet_gateway + ' dev ' + interface)
+  return route_cmds
+
+
+def GetMinMax(data):
+  """Finds min and max values in the given list of values"""
+  if not data:
+    return 0.0, 0.0
+  min_value = float(data[0])
+  max_value = float(data[0])
+  for item in data:
+    value = float(item)
+    if value > max_value:
+      max_value = value
+    if value < min_value:
+      min_value = value
+  return min_value, max_value
+
+
+def GetBaseOs(osdata):
+  """Finds the base os name to use
+
+  Args:
+    osdata: json representation of os information of an image.
+
+  Returns:
+    Short name of the os name
+  """
+  if 'name' in osdata and osdata['name'] is not None:
+    osname = osdata['name'].lower()
+    if 'window' in osname:
+      return WINDOWS
+    elif 'red' in osname or 'rhel' in osname:
+      return REDHAT
+    elif 'centos' in osname:
+      return CENTOS
+    elif 'debian' in osname:
+      return DEBIAN
+    elif 'ubuntu' in osname:
+      return UBUNTU
+  return UNKNOWN
+
+
+def GetGen(account):
+  """Creates a ibmcloud access object """
+  gen = ibmcloud.IbmCloud(account=account[0], apikey=account[1], verbose=False, \
```

Comment:
I'm a little leary of seeing references to tuple positions.  Can you do this?

```
@dataclasses.dataclass(frozen=True):
class Account:
   name:str
   apikey:str
```

then it is a lot easier to following:

```
def GetGen(account:Account) -> ?:
    get =   IbmCloud(account=account.name, apikey=account.apikey, ....)

Reply:
thanks for this tip here too, I don't like to use tuple indexes either ;)